libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -474,4 +476,53 @@ export class LicenseServiceService {
     }
     return verification.data
   }
+
+  async createBarcode(
+    { nationalId }: User,
+    genericUserLicense: GenericUserLicense,
+  ) {
+    const code = randomUUID()
+    const licenseType = genericUserLicense.license.type
+    const client = await this.getClient<typeof licenseType>(licenseType)
+    let extraData: LicenseVerifyExtraDataResult<LicenseType> | undefined
+
+    if (client?.verifyExtraData) {
+      extraData = await client.verifyExtraData({
+        licenseId: genericUserLicense.payload?.metadata?.licenseId,
+        nationalId,
+      })
+
+      if (!extraData) {
+        const msg = `Unable to verify extra data for license: ${licenseType}`
+        this.logger.warn(msg, { category: LOG_CATEGORY })
+
+        throw new InternalServerErrorException(msg)
+      }
+    }
+
+    // Create a token with the version, license type and a server reference (Redis key) code
+    const createTokenPromise = this.tokenService.createToken(
+      {
+        v: '2',
+        t: genericUserLicense.license.type,
+        c: code,
+      },
+      { expiresIn: '1m' },
+    )
+
+    // Store license data in cache so that we can fetch data quickly in the verify resolver method
+    const redisPromise = this.cacheManager.set(
+      code,
```

Comment:
Duplicate code generation would be very unfortunate (seeing a random person's drivers license). The most secure way to avoid it is using redis SETNX, which is not exposed in our Cache interface.

Not sure if this is worth spending time on, as it is exceedingly unlikely to happen. With 62^10 combinations, 1m expiration and not every barcode being scanned.

Reply:
Updated to 16 char to ease your mind

<img width="800" alt="image" src="https://github.com/island-is/island.is/assets/112904566/9648df76-00b5-4fa9-933b-fb7f1c1e7e48">